### ISSUE TITLE
Caching an array leads to `undefined method ´ response_headers'`

### ISSUE THREAD

**Issue by mikevoets** [flexirest/flexirest#58]
The API which I am talking to, always returns an object starting with a 'data' key, with its value either an array of record hashes or an single hash containing a record.

With a proxy class, I try to set the response body so that the body is the 'data' value (so either an array of hashes or single hash).

I have the following setup:
```
# service/base.rb
module Service
  class Unwrapper < Flexirest::ProxyBase
    get %r(.*) do
      response = passthrough
      translate(response) do |body|
        body["data"]
      end
    end
  end

  class Base < Flexirest::Base
    proxy Service::Unwrapper
    ...  
  end 
end
```
This works fine. However, only when I don't turn on caching: `Flexirest::Base.perform_caching = true`. When I turn on caching, and I get an array back from the API:
```
{"data": [
    ...
]}
```
It will put the array in the cache, and the next time when it gets a cache hit, the following exception occurs:
```
NoMethodError: undefined method `response_headers' for #<Flexirest::CachedResponse:0x000000091671b0>
	from /home/mike/.rbenv/versions/1.9.3-p448/lib/ruby/gems/1.9.1/gems/activesupport-3.2.22.2/lib/active_support/core_ext/object/try.rb:36:in `try'
	from /home/mike/.rbenv/versions/1.9.3-p448/lib/ruby/gems/1.9.1/gems/flexirest-1.3.27/lib/flexirest/result_iterator.rb:10:in `initialize'
	from /home/mike/.rbenv/versions/1.9.3-p448/lib/ruby/gems/1.9.1/gems/flexirest-1.3.27/lib/flexirest/caching.rb:107:in `new'
	from /home/mike/.rbenv/versions/1.9.3-p448/lib/ruby/gems/1.9.1/gems/flexirest-1.3.27/lib/flexirest/caching.rb:107:in `result'
	from /home/mike/.rbenv/versions/1.9.3-p448/lib/ruby/gems/1.9.1/gems/flexirest-1.3.27/lib/flexirest/request.rb:382:in `handle_cached_response'
	from /home/mike/.rbenv/versions/1.9.3-p448/lib/ruby/gems/1.9.1/gems/flexirest-1.3.27/lib/flexirest/request.rb:401:in `handle_response'
	from /home/mike/.rbenv/versions/1.9.3-p448/lib/ruby/gems/1.9.1/gems/flexirest-1.3.27/lib/flexirest/request.rb:213:in `block (2 levels) in call'
```

I tested on both development and production environment. This seems only to fail with arrays, not single hashes.

**Comment by andyjeffries** [comment]
Didn't mean to mark it as fixed. I have an attempted fix for this problem in commit 213a784, but I can't get an RSpec test to fail for it.

Can you please try replacing it in your Gemfile and see if it resolves it.

```
gem "flexirest", github: "flexirest/flexirest", ref: "213a784"
```

I'm not happy the fix going out without any spec coverage, but it's an innocuous change so I'm less worried than usual.

**Comment by mikevoets** [comment]
The fix works, thanks for your time. :-)

**Comment by andyjeffries** [comment]
Thanks, released as v1.3.30

**Comment by andyjeffries** [comment]
And you're welcome!
